app/scripts/views/mixins/avatar-mixin.js

```diff
@@ -7,81 +7,58 @@
 'use strict';
 
 define([
-], function () {
+  'models/profile-image'
+], function (ProfileImage) {
 
   return {
     initialize: function (options) {
       this.notifications = options.notifications;
     },
 
-    // Attempt to load a profile image from the profile server
-    _fetchProfileImage: function (account) {
+    displayProfileImage: function (account, wrapperClass) {
       var self = this;
-
-      return account.getAvatar()
-        .then(function (result) {
-          if (result && result.avatar && result.id) {
-            self.logEvent(self.className + '.profile_image_shown');
-          } else {
-            self.logEvent(self.className + '.profile_image_not_shown');
-          }
-
-          return result;
-        }, function (err) {
-          self.logEvent(self.className + '.profile_image_not_shown');
-          // Failures to load a profile image are not displayed in the ui
-          // so log the error here to make sure it's in metrics.
-          self.logError(err);
-          throw err;
-        });
-    },
-
-    _displayProfileImage: function (account, wrapperClass) {
-      var self = this;
-      if (! account) {
-        return;
-      }
-
       if (! wrapperClass) {
         wrapperClass = '.avatar-wrapper';
       }
 
-      // If the account doesn't have a profile image URL cached it
-      // probably doesn't have one, so show the default image immediately
-      // while we check for a real image
-      if (! account.has('profileImageUrl')) {
+      // We'll optimize the UI for the case that the account
+      // doesn't have a profile image if it's not cached
+      if (self._shouldShowDefaultProfileImage(account)) {
         self.$(wrapperClass).addClass('with-default');
       }
 
-      return this._fetchProfileImage(account)
-        .then(function (result) {
-          if (result && result.avatar) {
-            self.$(wrapperClass).append(new Image());
-            self.$(wrapperClass + ' img').attr('src', result.avatar);
-            self.$(wrapperClass).removeClass('with-default');
-          } else {
+      return account.fetchCurrentProfileImage(account)
```

Comment:
Not for this PR; I feel like a little bit of MVC could be used to make this code clearer. fetchCurrentProfileImage would immediately return a ProfileImageModel. A corresponding ProfileImageView would be created & rendered, by default it renders the default image. When the XHR request finishes, the ProfileImageModel is updated, which notifies the ProfileImageView, which re-renders itself.